Src/AutoMoq/MockVirtualMethodsCommand.cs

```diff
@@ -125,15 +126,29 @@ private static IEnumerable<MethodInfo> GetConfigurableMethods(Type type)
 
             // Skip properties that have both getters and setters to not interfere
             // with other post-processors in the chain that initialize them later.
-            var getMethods = type.GetProperties()
-                                    .Where(p => p.GetGetMethod() != null &&
-                                                p.GetSetMethod() != null)
-                                    .Select(p => p.GetGetMethod());
-            methods = methods.Except(getMethods);
+            methods = RemoveGetterSetterMethods(type, methods);
 
             return methods.Where(CanBeConfigured);
         }
 
+        private static IEnumerable<MethodInfo> RemoveGetterSetterMethods(Type type, IEnumerable<MethodInfo> methods)
+        {
+            var getterSetterMethods = GetProperties(type)
+                .Where(p => p.GetGetMethod() != null &&
+                            p.GetSetMethod() != null)
+                .Select(p => p.GetGetMethod());
+            methods = methods.Except(getterSetterMethods);
+            return methods;
+        }
+
+        private static IEnumerable<PropertyInfo> GetProperties(Type type)
```

Comment:
I'd suggest to encapsulate all this logic inside the extension method you introduced above. We can name it `GetAllProperties()` and perform the necessary checks there. Otherwise, logic becomes a bit split across the different files 😕

Reply:
done